examples/lines_bars_and_markers/curve_error_band.py

```diff
@@ -0,0 +1,82 @@
+"""
+=====================
+Curve with error band
+=====================
+
+This example illustrates how to draw an error band around a parametrized curve.
+
+A parametrized curve x(t), y(t) can directly be drawn using `~.Axes.plot`.
+"""
+# sphinx_gallery_thumbnail_number = 2
+
+import numpy as np
+from scipy.interpolate import splprep, splev
+
+import matplotlib.pyplot as plt
+from matplotlib.path import Path
+from matplotlib.patches import PathPatch
+
+N = 400
+phi = np.linspace(0, 2 * np.pi, N)
+r = 0.5 + np.cos(phi)
+x, y = r * np.cos(phi), r * np.sin(phi)
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+plt.show()
+
+#############################################################################
+# An error band can be used to indicate the uncertainty of the the curve.
+# In this example we assume that the error can be given as a scalar *err*
+# that describes the uncertainty perpendicular to the curve in every point.
+#
+# We visualize this error as a colored band around the path using a
+# `.PathPatch`. The patch is created from two path segments *(xp, yp)*, and
+# *(xn, yn)* that are shifted by +/- *err* perpendicular to the curve *(x, y)*.
+#
+# Note: This method of using a `.PathPatch` is suited to arbitrary curves in
+# 2D. If you just have a standard y-vs.-x plot, you can use the simpler
+# `~.Axes.fill_between` method (see also
+# :doc:`/gallery/lines_bars_and_markers/fill_between_demo`).
+
+err = 0.05 * np.sin(2*phi)**2 + 0.04 + 0.02 * np.cos(9*phi + 2)
```

Comment:
This is an arbitrary choice of error, right? Should get a comment saying so.

Reply:
Not sure this is necessary. Almost all our examples use arbitrary data. But done.

Reply:
Sure, but this example is about adding an error band, so it _might_ appear as if this is something related to adding the curve, and not just a random error.